### DISCUSSION TITLE
Hyperparameter tuning of custom components

### DISCUSSION THREAD

**Discussion by shrinidhin** [explosion/spaCy#10447] [Help: Coding & Implementations] [answered]
Hello!I am currently working on a project for building a custom entity based sentiment analyzer. I have built and trained my custom components on a set of data. I would now like to tune the hyperparameters to get the best set of parameters for the mode. I understand that we can use Weights and Biases for building a sweep to tune hyperparamters. However, I am a bit confused about which are the hyperparameters to be tuned for both NER component and a textcat model. I have gone through spacy docs but would appreciate any more inputs.
Additionally, I would like to know that once I have completed training and hyperparameter tuning, usually I combine train and dev set and perform one last training run with the final hyperparameters to get the final trained model. Can I do the same for custom trained spacy components as well? Thank you!

**Comment by thomashacker** [comment, marked as answer]
Hello, 
Yes, you can combine the train and dev dataset to train custom components. In terms of what the most important parameters to change are: It really depends on the data you're working with. We're also currently working on a FAQ for this particular topic to help users get a better feeling/understanding about hyperparameters/sections in general. We don't have a detailed explanation about every hyperparameter for every pipeline in our docs but if you're interested you can look into the [NER](https://github.com/explosion/spaCy/blob/master/spacy/pipeline/ner.pyx) and [Textcat](https://github.com/explosion/spaCy/blob/master/spacy/pipeline/textcat.py) component code to get a more detailed insight into how the parameters influence the model/training and decide which suits best for your use-case based on that.

**Reply by shrinidhin** [reply]
Thanks a lot!

**Reply by Sumit5194** [reply]
did hyperparameter tuning in your Project?

**Reply by Sumit5194** [reply]
i am also looking for solution to do  hyperparameter tuning. I have build custom blank "en" ner model. i have trained but now i want to do hyperparameter tuning for same model. how to do it? @shrinidhin, @thomashacker  if you have any lead please let me know.

**Reply by polm** [reply]
Hey, please don't reply to old threads and @ people asking for help - it's better to open a new thread with your question.

The Hyperparameter FAQ mentioned above was posted here, and should be a good starting point for you or anyone else who comes here later. 

https://github.com/explosion/spaCy/discussions/10625
